File: ios/WellnessCompanion/UI/Journal/JournalListScreen.swift
import SwiftUI

struct JournalListScreen: View {
    @EnvironmentObject private var container: AppContainer
    @Environment(\.openReflection) private var openReflection
    
    @State private var entries: [JournalSummary] = []
    @State private var threads: [NarrativeThread] = []
    @State private var synthesizedInsight: String? = nil
    @State private var isSynthesizing = false
    
    var body: some View {
        libraryContent
            .onAppear(perform: refresh)
    }
    
    private var libraryContent: some View {
        ScrollView(showsIndicators: false) {
            VStack(alignment: .leading, spacing: 50) {
                
                // 1. High-Fidelity Hero Image
                VStack(alignment: .leading, spacing: 20) {
                    AsyncImage(url: URL(string: WellnessContentProvider.libraryHero)) { image in
                        image.resizable()
                            .aspectRatio(contentMode: .fill)
                            .frame(height: 240)
                            .clipShape(RoundedRectangle(cornerRadius: 48, style: .continuous))
                            .overlay(alignment: .bottomLeading) {
                                VStack(alignment: .leading, spacing: 4) {
                                    Text("REFLECTIONS").miniCaps().foregroundStyle(Color.white.opacity(0.6))
                                    Text("Library").font(.system(size: 44, weight: .black, design: .rounded)).foregroundStyle(.white)
                                }
                                .padding(32)
                            }
                    } placeholder: {
                        RoundedRectangle(cornerRadius: 48).fill(.white.opacity(0.05)).frame(height: 240)
                    }
                    
                    Text(WellnessContentProvider.attribution)
                        .font(.system(size: 8)).miniCaps().foregroundStyle(.white.opacity(0.2))
                        .padding(.leading, 12)
                }
                .padding(.top, 20)
                
                // 2. Intelligence Layer: Synthesis Card
                if !entries.isEmpty {
                    synthesisSection
                }
                
                if !threads.isEmpty {
                    VStack(alignment: .leading, spacing: 24) {
                        Text("Themes").sectionHeader()
                        ScrollView(.horizontal, showsIndicators: false) {
                            HStack(spacing: 20) {
                                ForEach(threads) { thread in
                                    VStack(alignment: .leading, spacing: 14) {
                                        Image(systemName: "sparkles").font(.title2).foregroundStyle(.cyan)
                                        VStack(alignment: .leading, spacing: 0) {
                                            Text(thread.label).font(.headline).foregroundStyle(.white)
                                            Text("\(thread.entryCount) entries").font(.caption).foregroundStyle(.white.opacity(0.4))
                                        }
                                    }
                                    .padding(24)
                                    .frame(width: 150, height: 130, alignment: .topLeading)
                                    .background(.white.opacity(0.08))
                                    .clipShape(RoundedRectangle(cornerRadius: 32, style: .continuous))
                                }
                            }
                        }
                    }
                }
                
                VStack(alignment: .leading, spacing: 24) {
                    Text("Timeline").sectionHeader()
                    if entries.isEmpty {
                        VStack(spacing: 16) {
                            Image(systemName: "book.closed")
                                .font(.system(size: 40))
                                .foregroundStyle(.white.opacity(0.2))
                            Text("Your story begins here.")
                                .font(.system(size: 20, weight: .medium, design: .serif))
                                .foregroundStyle(.white.opacity(0.4))
                        }
                        .frame(maxWidth: CGFloat.infinity)
                        .padding(.top, 60)
                    } else {
                        ForEach(entries) { summary in
                            Button {
                                openReflection(summary.id, nil)
                            } label: {
                                LiquidEntryRow(summary: summary)
                            }
                            .buttonStyle(.plain)
                        }
                    }
                }
                
                Spacer(minLength: 150)
            }
            .padding(.horizontal, 28)
        }
        .overlay(alignment: .bottomTrailing) {
            // Floating Plus Button
            Button {
                openReflection(nil, nil)
            } label: {
                Circle()
                    .fill(Color.wellnessAccent)
                    .frame(width: 72, height: 72)
                    .shadow(color: Color.wellnessAccent.opacity(0.3), radius: 20, y: 10)
                    .overlay(Image(systemName: "plus").font(.system(size: 30, weight: .bold)).foregroundStyle(.white))
            }
            .padding(28)
            .padding(.bottom, 80)
        }
    }
    
    private var synthesisSection: some View {
        VStack(alignment: .leading, spacing: 20) {
            HStack {
                Text("Synthesized Insight").sectionHeader()
                Spacer()
                if isSynthesizing {
                    ProgressView().tint(.cyan).scaleEffect(0.8)
                } else {
                    Button { runSynthesis() } label: {
                        Image(systemName: "arrow.clockwise").font(.caption.bold()).foregroundStyle(.cyan)
                    }
                }
            }
            
            VStack(alignment: .leading, spacing: 16) {
                if let insight = synthesizedInsight {
                    Text(insight)
                        .font(.system(size: 18, weight: .medium, design: .serif))
                        .italic()
                        .foregroundStyle(.white)
                        .lineSpacing(6)
                        .transition(.opacity.combined(with: .move(edge: .top)))
                } else if !isSynthesizing {
                    Text("Deep-dive analysis of your last 3 reflections.")
                        .font(.caption).foregroundStyle(.white.opacity(0.4))
                } else {
                    Text("Finding the hidden thread…")
                        .font(.caption).foregroundStyle(.cyan.opacity(0.6))
                }
            }
            .padding(32)
            .frame(maxWidth: .infinity, alignment: .leading)
            .background(.white.opacity(0.08))
            .clipShape(RoundedRectangle(cornerRadius: 40, style: .continuous))
            .overlay(RoundedRectangle(cornerRadius: 40, style: .continuous).stroke(.cyan.opacity(0.2), lineWidth: 1))
        }
    }

    private func refresh() {
        entries = container.journalStore.fetchSummaries()
        threads = container.narrativeStore.activeThreads()
        if synthesizedInsight == nil && entries.count >= 2 {
            runSynthesis()
        }
    }
    
    private func runSynthesis() {
        guard let engine = container.reflectionEngine, entries.count >= 2 else { return }
        isSynthesizing = true
        Task {
            // Fetch the full entries for synthesis
            let fullEntries = entries.prefix(3).compactMap { container.journalStore.fetchById($0.id) }
            let insight = await engine.synthesizeInsight(entries: fullEntries)
            await MainActor.run {
                withAnimation(.spring()) {
                    synthesizedInsight = insight
                    isSynthesizing = false
                }
            }
        }
    }
}

File: app/src/main/java/com/wellness/companion/ui/journal/JournalListScreen.kt
package com.wellness.companion.ui.journal

import androidx.compose.animation.*
import androidx.compose.animation.core.*
import androidx.compose.foundation.*
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.*
import androidx.compose.foundation.shape.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.*
import androidx.compose.ui.draw.*
import androidx.compose.ui.graphics.*
import androidx.compose.ui.layout.*
import androidx.compose.ui.platform.*
import androidx.compose.ui.text.font.*
import androidx.compose.ui.text.style.*
import androidx.compose.ui.unit.*
import coil.compose.AsyncImage
import com.wellness.companion.data.db.JournalSummary
import com.wellness.companion.di.AppContainer
import com.wellness.companion.ui.WellnessContentProvider
import com.wellness.companion.ui.components.LiquidAura
import com.wellness.companion.ui.theme.WellnessPalette
import kotlinx.coroutines.launch

@Composable
fun JournalListScreen(
    container: AppContainer,
    onOpen: (Long) -> Unit,
    onOpenThread: (Long, String) -> Unit,
    contentPadding: PaddingValues = PaddingValues(0.dp)
) {
    val viewModel: JournalListViewModel = androidx.lifecycle.viewmodel.compose.viewModel(
        factory = com.wellness.companion.di.ViewModelFactories.journalList(container)
    )
    val state by viewModel.state.collectAsState()
    
    var synthesizedInsight by remember { mutableStateOf<String?>(null) }
    var isSynthesizing by remember { mutableStateOf(false) }

    LaunchedEffect(state.entries.size) {
        if (state.entries.size >= 2 && synthesizedInsight == null) {
            isSynthesizing = true
            val fullEntries = container.journalRepository.fetchRecentEntries(3)
            synthesizedInsight = container.reflectionEngine?.synthesizeInsight(fullEntries)
            isSynthesizing = false
        }
    }

    Box(Modifier.fillMaxSize()) {
        LiquidAura()

        Column(
            modifier = Modifier
                .fillMaxSize()
                .statusBarsPadding()
                .padding(contentPadding)
                .verticalScroll(rememberScrollState())
                .padding(horizontal = 28.dp)
        ) {
            Spacer(Modifier.height(20.dp))

            // 1. High-Fidelity Hero Image
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(240.dp)
                    .clip(RoundedCornerShape(48.dp))
            ) {
                AsyncImage(
                    model = WellnessContentProvider.LIBRARY_HERO,
                    contentDescription = null,
                    contentScale = ContentScale.Crop,
                    modifier = Modifier.fillMaxSize()
                )
                Box(
                    Modifier
                        .fillMaxSize()
                        .background(
                            Brush.verticalGradient(
                                colors = listOf(Color.Transparent, Color.Black.copy(alpha = 0.6f))
                            )
                        )
                )
                Column(
                    Modifier
                        .align(Alignment.BottomStart)
                        .padding(32.dp)
                ) {
                    Text("REFLECTIONS", style = MaterialTheme.typography.labelSmall, color = Color.White.copy(alpha = 0.7f), letterSpacing = 2.sp)
                    Text("Library", style = MaterialTheme.typography.headlineLarge, fontWeight = FontWeight.Black, color = Color.White)
                }
            }
            
            Text(
                text = WellnessContentProvider.ATTRIBUTION,
                style = MaterialTheme.typography.labelSmall,
                color = Color.White.copy(alpha = 0.2f),
                modifier = Modifier.padding(top = 8.dp, start = 12.dp)
            )

            Spacer(Modifier.height(40.dp))

            // 2. Intelligence Layer: Synthesis Card
            if (state.entries.isNotEmpty()) {
                SynthesisSection(insight = synthesizedInsight, isSynthesizing = isSynthesizing) {
                    isSynthesizing = true
                    val fullEntries = container.journalRepository.fetchRecentEntries(3)
                    synthesizedInsight = container.reflectionEngine?.synthesizeInsight(fullEntries)
                    isSynthesizing = false
                }
                Spacer(Modifier.height(50.dp))
            }

            // 3. Themes
            if (state.threads.isNotEmpty()) {
                Text("THEMES", style = MaterialTheme.typography.labelSmall, fontWeight = FontWeight.Black, color = Color.White.copy(alpha = 0.5f))
                Spacer(Modifier.height(20.dp))
                LazyRow(horizontalArrangement = Arrangement.spacedBy(16.dp)) {
                    items(state.threads) { thread ->
                        Surface(
                            onClick = { onOpenThread(thread.id, thread.label) },
                            modifier = Modifier.size(150.dp, 130.dp),
                            color = Color.White.copy(alpha = 0.08f),
                            shape = RoundedCornerShape(32.dp),
                            border = BorderStroke(1.dp, Color.White.copy(alpha = 0.1f))
                        ) {
                            Column(Modifier.padding(24.dp), verticalArrangement = Arrangement.spacedBy(14.dp)) {
                                Icon(Icons.Default.AutoAwesome, null, tint = Color.Cyan, modifier = Modifier.size(24.dp))
                                Column {
                                    Text(thread.label, fontWeight = FontWeight.Bold, color = Color.White, maxLines = 1, overflow = TextOverflow.Ellipsis)
                                    Text("${thread.entryCount} entries", style = MaterialTheme.typography.labelSmall, color = Color.White.copy(alpha = 0.4f))
                                }
                            }
                        }
                    }
                }
                Spacer(Modifier.height(50.dp))
            }

            // 4. Timeline
            Text("TIMELINE", style = MaterialTheme.typography.labelSmall, fontWeight = FontWeight.Black, color = Color.White.copy(alpha = 0.5f))
            Spacer(Modifier.height(20.dp))
            state.entries.forEach { summary ->
                RecentActivityRow(summary, onOpen)
                Spacer(Modifier.height(16.dp))
            }

            Spacer(Modifier.height(150.dp))
        }

        // Floating Plus Button
        FloatingActionButton(
            onClick = { onOpen(-1L) },
            modifier = Modifier
                .align(Alignment.BottomEnd)
                .padding(28.dp)
                .padding(bottom = 80.dp),
            containerColor = WellnessPalette.LiquidIndigo,
            contentColor = Color.White,
            shape = CircleShape
        ) {
            Icon(Icons.Default.Add, null, modifier = Modifier.size(30.dp))
        }
    }
}

@Composable
private fun SynthesisSection(insight: String?, isSynthesizing: Boolean, onRefresh: suspend () -> Unit) {
    val scope = rememberCoroutineScope()
    Column(verticalArrangement = Arrangement.spacedBy(20.dp)) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Text("SYNTESIZED INSIGHT", style = MaterialTheme.typography.labelSmall, fontWeight = FontWeight.Black, color = Color.White.copy(alpha = 0.5f))
            Spacer(Modifier.weight(1f))
            if (isSynthesizing) {
                CircularProgressIndicator(Modifier.size(16.dp), color = Color.Cyan, strokeWidth = 2.dp)
            } else {
                IconButton(onClick = { scope.launch { onRefresh() } }) {
                    Icon(Icons.Default.Refresh, null, tint = Color.Cyan, modifier = Modifier.size(16.dp))
                }
            }
        }
        
        Surface(
            color = Color.White.copy(alpha = 0.08f),
            shape = RoundedRectangle(40.dp),
            border = BorderStroke(1.dp, Color.Cyan.copy(alpha = 0.2f)),
            modifier = Modifier.fillMaxWidth()
        ) {
            Box(Modifier.padding(32.dp)) {
                if (insight != null) {
                    Text(insight, style = MaterialTheme.typography.bodyLarge.copy(fontFamily = FontFamily.Serif), fontStyle = FontStyle.Italic, color = Color.White)
                } else {
                    Text(
                        if (isSynthesizing) "Finding the hidden thread…" else "Analysis of your last 3 reflections.",
                        style = MaterialTheme.typography.labelMedium,
                        color = if (isSynthesizing) Color.Cyan.copy(alpha = 0.6f) else Color.White.copy(alpha = 0.4f)
                    )
                }
            }
        }
    }
}

@Composable
private fun RecentActivityRow(summary: JournalSummary, onClick: (Long) -> Unit) {
    Surface(
        onClick = { onClick(summary.id) },
        color = Color.White.copy(alpha = 0.05f),
        shape = RoundedRectangle(32.dp),
        modifier = Modifier.fillMaxWidth()
    ) {
        Row(Modifier.padding(20.dp), verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(20.dp)) {
            Icon(Icons.Default.TextSnippet, null, tint = Color.Cyan, modifier = Modifier.size(24.dp))
            Column {
                Text(summary.title, fontWeight = FontWeight.Bold, color = Color.White)
                Text("${summary.wordCount} words", style = MaterialTheme.typography.labelSmall, color = Color.White.copy(alpha = 0.4f))
            }
            Spacer(Modifier.weight(1f))
            Icon(Icons.Default.ChevronRight, null, tint = Color.White.copy(alpha = 0.2f))
        }
    }
}

private fun RoundedRectangle(radius: Dp) = RoundedCornerShape(radius)
